File: maxhanna.client/src/services/datacontracts/trade/profit-data.ts
export class ProfitData {
  periodType?: 'daily' | 'weekly' | 'monthly';
  periodStart?: Date | null;
  periodEnd?: Date | null;
  startUsdc?: number;
  startBtc?: number;
  startBtcPriceUsdc?: number;
  endUsdc?: number;
  endBtc?: number;
  endBtcPriceUsdc?: number;
  profitUsdc?: number;
  cumulativeProfitUsdc?: number;

  constructor(data?: Partial<ProfitData>) {
    if (data) {
      this.periodType = data.periodType;
      this.periodStart = data.periodStart ? new Date(data.periodStart) : null;
      this.periodEnd = data.periodEnd ? new Date(data.periodEnd) : null;
      this.startUsdc = data.startUsdc;
      this.startBtc = data.startBtc;
      this.startBtcPriceUsdc = data.startBtcPriceUsdc;
      this.endUsdc = data.endUsdc;
      this.endBtc = data.endBtc;
      this.endBtcPriceUsdc = data.endBtcPriceUsdc;
      this.profitUsdc = data.profitUsdc;
      this.cumulativeProfitUsdc = data.cumulativeProfitUsdc;
    }
  }
}

File: maxhanna.Server/Controllers/DataContracts/Crypto/ProfitData.cs
public class ProfitData
{
	public required string PeriodType { get; set; } // "daily", "weekly", or "monthly"
	public DateTime? PeriodStart { get; set; }
	public DateTime? PeriodEnd { get; set; }
	public decimal StartUsdc { get; set; }
	public decimal StartBtc { get; set; }
	public decimal StartBtcPriceUsdc { get; set; }
	public decimal EndUsdc { get; set; }
	public decimal EndBtc { get; set; }
	public decimal EndBtcPriceUsdc { get; set; }
	public decimal ProfitUsdc { get; set; }
	public decimal CumulativeProfitUsdc { get; set; }
	public decimal AbsoluteProfitUsdc { get; set; }
}
